hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeParentRecordHandler.java

```diff
@@ -85,7 +92,18 @@ public void finalizeParentRecord(@NonNull final AccountID payer, @NonNull final
         }
 
         // Hbar changes from transaction including staking rewards
-        final var hbarChanges = hbarChangesFrom(writableAccountStore);
+        final Map<AccountID, Long> hbarChanges;
+        try {
+            hbarChanges = hbarChangesFrom(writableAccountStore);
+        } catch (HandleException e) {
+            if (e.getStatus() == FAIL_INVALID) {
+                logHbarFinalizationFailInvalid(
+                        payer,
+                        context.userTransactionRecordBuilder(SingleTransactionRecordBuilder.class),
+                        writableAccountStore);
+            }
+            throw e;
+        }
```

Comment:
Since we are adding logs we could also add to `modifiedTokenRelBalance != 0` also

Reply:
Hmm but it's possible to have non-zero net changes to token balances...e.g. on a `TokenBurn`, right?